Release notes for 1.28.6 and 1.29.2 releases (#17317)
istio/istio.io: content/en/news/releases/1.29.x/announcing-1.29.2/index.md

---
title: Announcing Istio 1.29.2
linktitle: 1.29.2
subtitle: Patch Release
description: Istio 1.29.2 patch release.

> publishdate: 2026-03-10

Reviewer: Publish date should be today:
```suggestion
publishdate: 2026-04-13
```

release: 1.29.2
aliases:
    - /news/announcing-1.29.2
---

This release contains security fixes. This release note describes what's different between Istio 1.29.1 and 1.29.2.

{{< relnote >}}

## Changes

- **Added** Helm v4 (server-side apply) support. Fixed a webhook `failurePolicy` field ownership
  conflict that caused `helm upgrade` with SSA to fail.
  ([Issue #58302](https://github.com/istio/istio/issues/58302)) ([Issue #59367](https://github.com/istio/istio/issues/59367))

> - **Fixed** istiod errors on startup when a CRD version greater than the maximum supported version is installed on a cluster. TLS route versions v1.4 and below are supported; v1.5 and above will be ignored.

Reviewer: `TLSRoute` is the K8s resource name and should be backticked:
```suggestion
- **Fixed** istiod errors on startup when a CRD version greater than the maximum supported version is installed on a cluster. `TLSRoute` versions v1.4 and below are supported; v1.5 and above will be ignored.
```

  ([Issue #59443](https://github.com/istio/istio/issues/59443))

- **Fixed** multi-cluster installations trying to validate the wrong trust domain when the
  control plane does not have an updated `istio-reader` `ClusterRole`, failing to read the

>   trust domain from the remote ConfigMap. Now, istiod will fall back to using the

Reviewer: `ConfigMap` is a K8s resource and should be backticked:
```suggestion
  trust domain from the remote `ConfigMap`. Now, istiod will fall back to using the
```

  trust domain specified in the local mesh config until it can read the remote one.
  ([Issue #59474](https://github.com/istio/istio/issues/59474))

> - **Fixed** applying multiple `VirtualService` for the same hostname to waypoints.

Reviewer: Same as 1.28.6 — needs "resources":
```suggestion
- **Fixed** applying multiple `VirtualService` resources for the same hostname to waypoints.
```

  ([Issue #59483](https://github.com/istio/istio/issues/59483))

- **Fixed** an issue where `istioctl` incorrectly reported an error on `EnvoyFilter` with `REPLACE` operation on `VIRTUAL_HOST`.
  ([Issue #59495](https://github.com/istio/istio/issues/59495))

> - **Fixed** a bug when E/W gateway occasionally routed HBONE connections to a wrong service due to

Reviewer: "when" should be "where" (describing a condition, not a time), and missing article "the":
```suggestion
- **Fixed** a bug where the E/W gateway occasionally routed HBONE connections to a wrong service due to
```

  incorrect connection pooling in Envoy.
  ([Issue #58630](https://github.com/istio/istio/issues/58630))

- **Fixed** `serviceAccount` matcher regex in `AuthorizationPolicy` to properly quote the service account name, allowing for correct matching of service accounts with special characters in their names.
  ([Issue #59700](https://github.com/istio/istio/issues/59700))

> - **Fixed** an issue where all Gateways were restarted after istiod was restarted.

Reviewer: `Gateways` refers to the K8s Gateway resource and should be backticked:
```suggestion
- **Fixed** an issue where all `Gateways` were restarted after istiod was restarted.
```

  ([Issue #59709](https://github.com/istio/istio/issues/59709))

> - **Fixed** `TLSRoute` hostnames not being constrained to the intersection with the Gateway listener hostname.

Reviewer: `Gateway` is a K8s resource name and should be backticked:
```suggestion
- **Fixed** `TLSRoute` hostnames not being constrained to the intersection with the `Gateway` listener hostname.
```

  Previously, a `TLSRoute` with a broad hostname (e.g. `*.com`) attached to a listener with a narrower hostname
  (e.g. `*.example.com`) would incorrectly match the full route hostname instead of only the intersection
  (`*.example.com`), as required by the Gateway API spec.
  ([Issue #59229](https://github.com/istio/istio/issues/59229))

- **Fixed** JWKS URI CIDR blocking by using a custom control function in a custom `DialContext`.
  The control function filters connections after DNS resolution but before dialing, allowing

>   the block to follow redirects and issuer discovery path. This also preserves features

Reviewer: Missing article — "the issuer discovery path":
```suggestion
  the block to follow redirects and the issuer discovery path. This also preserves features
```
